db/migrate/20230921181155_add_territory_id_to_services.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class AddTerritoryIdToServices < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :services, :territory, foreign_key: true, index: true
+
+    # Ce script a pour vocation de créer les services en spécifiant leur territoire, sur la base des services existant
+    # Pour chaque service, identifier les territoires liés
+    # Pour chaque service:
+    #  - Pour chaque territoire du service:
+    #     - Dupliquer le service dans le territoire
+    #     - Mettre à jour tous les agents du territoire avec le nouveau service
+    #     - Mettre à jour tous les motifs du territoire avec le nouveau service
+    #
+    # Supprimer tous les services sans TerritoryID
+
+    services = Service.joins(
+      agents: { organisations: :territory }
+    ).select(
+      :id, :name, :short_name, "ARRAY_AGG(DISTINCT territories.id) as territories_ids"
+    ).group(
+      :id, :name, :short_name
+    )
+
+    Service.transaction do
+      services.each do |service|
+        service.territories_ids.each do |territory_id|
+          new_service = service.dup
+          new_service.territory_id = territory_id
+
+          # This is to avoid issues with the unique validation on name and short_name columns
+          service.update!(name: "old_#{service.name}", short_name: "old_#{service.short_name}")
+          new_service.save!
+
+          Agent.joins(organisations: :territory).where(service_id: service.id, organisations: { territory_id: territory_id }).update_all(service_id: new_service.id)
+          Motif.joins(organisation: :territory).where(service_id: service.id, organisation: { territory_id: terr }).update_all(service_id: new_service.id)
+        end
```

Comment:
il y a une erreur `terr` -> `territory_id`
J'ai essayé de faire tourner la migration en local avec la base de donnée seedé et j'ai une erreur : 
```ActiveRecord::DeleteRestrictionError: Cannot delete record because of dependent motifs```
```/code/rdv-solidarites.fr/db/migrate/20230921181155_add_territory_id_to_services.rb:40:in `block in change'```